### ISSUE TITLE
Debugging why attributes appear in classes 

### ISSUE THREAD

**Issue by skinkie** [tefra/xsdata#734]
I am currently evaluating _why_ the NeTEx XSD adds the dated_vehicle_journey_ref to route_ref.

I can see that:
 1. order comes from PointInLinkSequence_VersionedChildStructure
 1. link_sequence_ref, projections, description come from PointInLinkSequenceGroup.

~Might this be a bug in the schema creation?~

https://github.com/NeTEx-CEN/NeTEx/blob/master/xsd/netex_framework/netex_genericFramework/netex_pointAndLinkSequence_version.xsd#L134~

Would it be possible to annotate where an attribute originates from in a tree like representation?

~Upon searching my hypothesis is now that "LinkSequenceRef" is to blame, which is used both as substitution group as it is as element ref.~

~When removing this subsitutiongroup, navigation_path_ref is removed, so I am confident there is a relationship...
https://github.com/NeTEx-CEN/NeTEx/blob/master/xsd/netex_part_1/part1_ifopt/netex_ifopt_path_support.xsd#L184~

```
@dataclass(unsafe_hash=True)
class PointInLinkSequenceVersionedChildStructure(VersionedChildStructure):
    """
    Type for a POINT IN LINK SEQUENCE i.e. a Node of a graph.

    :ivar dated_vehicle_journey_ref:
    :ivar dated_special_service_ref:
    :ivar special_service_ref:
    :ivar template_service_journey_ref:
    :ivar service_journey_ref:
    :ivar dead_run_ref:
    :ivar vehicle_journey_ref:
    :ivar navigation_path_ref:
    :ivar service_journey_pattern_ref:
    :ivar service_pattern_ref:
    :ivar dead_run_journey_pattern_ref:
    :ivar journey_pattern_ref:
    :ivar timing_pattern_ref:
    :ivar route_ref:
    :ivar link_sequence_ref:
    :ivar projections: PROJECTIONs of POINT.
    :ivar description: Further Description of a POINT IN LINK SEQUENCE
    :ivar order: Order of POINT within parent.
```

To summarize. I think there is a bug that introduces elements from a substitution group "LinkSequenceRef", based on every element that has the substitutionGroup "LinkSequenceRef".

**Comment by tefra** [comment]
I am a bit confused what attributes do you think are wrong?

Yeah the  class is getting pretty much every property above the link_sequence_ref because of substitution groups

<img width="1105" alt="Screenshot 2022-12-22 at 15 52 00" src="https://user-images.githubusercontent.com/2098915/209148759-0aebbd6e-b163-4385-8a34-ff4bf62ab171.png">

Yeah I think the origin metadata would be super helpful

**Comment by skinkie** [comment]
@tefra I really really wonder if there should be **any** property of the link_sequence_ref on the PointInLinkSequence?

From my understanding (and discussion with the NeTEx XSD developers) the following attributes are unexpected:
```
    :ivar dated_vehicle_journey_ref:
    :ivar dated_special_service_ref:
    :ivar special_service_ref:
    :ivar template_service_journey_ref:
    :ivar service_journey_ref:
    :ivar dead_run_ref:
    :ivar vehicle_journey_ref:
    :ivar navigation_path_ref:
    :ivar service_journey_pattern_ref:
    :ivar service_pattern_ref:
    :ivar dead_run_journey_pattern_ref:
    :ivar journey_pattern_ref:
    :ivar timing_pattern_ref:
    :ivar route_ref:
```

**Comment by skinkie** [comment]
@tefra the tool reports the correct attributes. Document below validates against the NeTEx schema. I'll go back upstream. For the part of this ticket to 'explain' where an attribute originates from, I'll would like to keep this feature request open.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<PublicationDelivery xmlns="http://www.netex.org.uk/netex" xmlns:gml="http://www.opengis.net/gml/3.2" version="ntx:1.1">
  <PublicationTimestamp>2022-12-23T09:56:19.513998</PublicationTimestamp>
  <ParticipantRef>NDOV</ParticipantRef>
  <Description>NeTEx export</Description>
  <dataObjects>
    <GeneralFrame id="GF" version="1">
      <members>
        <ScheduledStopPoint id="SSP1" version="1">
          <Name>SSP1</Name>
        </ScheduledStopPoint>
        <ScheduledStopPoint id="SSP2" version="1">
          <Name>SSP2</Name>
        </ScheduledStopPoint>
        <ServiceJourneyPattern id="SJP" version="1">
          <pointsInSequence>
            <StopPointInJourneyPattern id="SPIJP1" version="1" order="1">
              <ServiceJourneyPatternRef version="1" ref="SJP"/>
              <ScheduledStopPointRef version="1" ref="SSP1"/>
            </StopPointInJourneyPattern>
            <StopPointInJourneyPattern id="SPIJP2" version="1" order="2">
              <ScheduledStopPointRef version="1" ref="SSP2"/>
            </StopPointInJourneyPattern>
          </pointsInSequence>
        </ServiceJourneyPattern>
      </members>
    </GeneralFrame>
  </dataObjects>
</PublicationDelivery>
```

**Comment by tefra** [comment]
I am closing this one for now, I will gladly accept any contributions
